### DISCUSSION TITLE
using ase for loading datasets

### DISCUSSION THREAD

**Discussion by nisargjoshi10** [mir-group/nequip#190] [Q&A] [answered]
I'm curious if there's a reason for using just ase for loading simulations data? In my experience, I have been working with different simulations tools and have observed that `ase` doesn't load most of the datasets. For e.g. I have many simulations results from GROMACS and LAMMPS but all of them store the forces and trajectories either into `trr` format or a binary `dump` file and I think I've not successfully converted that data to load it into nequip.

If you have any suggestions with what could be simpler data conversions, I'm curious to find out simpler methods.

Thanks!

**Comment by Linux-cpp-lisp** [comment, marked as answer]
hi @nisargjoshi10 ,

ASE provides a large variety of parsers in a common interface, although of course not every format can be supported. In particular ASE has mature parsers for a wide variety of quantum chemistry codes, which are typically the source of training data. (It is very unusual to get training data from classical potentials such as those used in GROMACS.)

ASE does support LAMMPS output files, please see their documentation.

For formats not supported by ASE, you can find another parser or write your own, create `ase.Atoms` objects containing the relevant information, and write them out to `extxyz` format using `ase.io.write`. In general NequIP is not in the business of writing, maintaining, or debugging parsers and we are not currently considering adding any other formats or parsers to the code.
